lib/ansible/modules/cloud/amazon/lambda_bucket_event.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/python
+# (c) 2019, XLAB d.o.o <www.xlab.si>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: lambda_bucket_event
+short_description: Creates, updates or deletes S3 Bucket notification for lambda
+description:
+    - This module allows the management of AWS Lambda function bucket event mappings via the
+      Ansible framework. Use module M(lambda) to manage the lambda function itself, M(lambda_alias)
+      to manage function aliases and M(lambda_policy) to modify lambda permissions.
+notes:
+    - This module heavily depends on M(lambda_policy) as you need to allow C(lambda:InvokeFunction)
+       permission for your lambda function.
+version_added: "2.9"
+
+author:
+    - XLAB d.o.o. (@xlab-si)
+options:
+  event_name:
+    description:
+      - unique name for event notification on bucket
+    required: True
+    type: str
+  lambda_function_arn:
+    description:
+      - The ARN of the lambda function.
+    required: true
+    aliases: ['function_arn']
+    type: str
+  bucket_name:
+    description:
+      - S3 bucket name
+    required: True
+    type: str
+  state:
+    description:
+      - Describes the desired state.
+    required: true
+    default: "present"
+    choices: ["present", "absent"]
+    type: str
+  alias:
+    description:
+      - Name of the function alias. Mutually exclusive with C(version).
+    required: true
```

Comment:
```suggestion
    required: false
```
The arg_spec is correct for this